doc/manual/checker_library.md

```diff
@@ -85,3 +85,52 @@ traceability). The metadata produced by a check shall be included and
 described in its documentation, (e.g., if a check logs in metadata version and
 configuration of a simulator, the checker documentation shall describe the
 content of aforementioned metadata fields)._
+
+## Manifest File
+
+Each Checker Library needs to provide a manifest file with configuration
+information for the framework. The format of this file is aligned with the one
+for the [Result File (`*.xqar`)](file_formats.md#result-file-xqar) and follows
+the same specification. The only difference is that it may contain additional
+Report Modules, which are delivered as part of the Checker Library.
+
+### Example
+
+```xml
+<CheckerResults version="1.0.0">
+    <CheckerBundle name="CheckerBundle1" description="Description of the Checker Bundle" build_date="2024-03-26" version="1.0.0">
+        <Executable name="/usr/bin/python3"> <!-- apply to Linux only! Probably just "python"? -->
+            <Arg>my_fancy_script.py</Arg> <!-- Mandatory arguments, followed by the configuration.xml or input file-->
+            <Arg>--fancy-param=True</Arg>
+        </Executable>
+        <Param name="InputFile" value="myTrack.xodr"/> <!-- Global standard parameter for each module! -->
+        <Param name="CheckerBundleParameter" value="0.1"/>
+        <Checker checkerId="XsdChecker" description="Checks the schema validity">
+            <AddressedRule uid="asam.net:qc:1.0.0:xsd_schema_validity">
+            <MetaData name="XmlParser" value="Xerces-C++ 1.2.3"> <!-- Or part of CheckerBunde? Or both possible like parameter? -->
+        </Checker>
+        <Checker checkerId="RefLineChecker" description="Checks the reference line integrity">
+            <AddressedRule uid="asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists">
+            <AddressedRule uid="example.com:::custom_rule">
+            <Param name="CheckerParameter" value="123"/> 
+        </Checker>
+    </CheckerBundle>
+    <ReportModule name="ReportModule" description="Description of the Report Module" build_date="2024-03-26" version="1.0.0">
+        <Executable name="text_report"/> <!-- standard executable without additional arguments, except the configuration.xml or result file -->
+        <Param name="InputFile" value="myTrack.xodr"/> <!-- Global standard parameter for each module! -->
+    </ReportModule>    
+</CheckerResults>
+```
+
+### Open Questions
+
+- How does the framework find this Manifest file and the appropriate Checker
+  Bundles and Report Modules?
+- Is a `<ReportModule/>` from the technical point-of-view the same as a
+  `<CheckerBundle/>` just without Checkers? Use the same element?
+  --> rename to `<Module/>`?
+- Path of executable?
+  - (bin?) directory of Checker Library
+  - can be found in $PATH
+  - global path ;-?
+- (...)
```

Comment:
I think this question is simply delegated to the operating system. We do not consider what the user wrote in the attribute `program="..."`. It will be just processed by OS -> sub process. This will perform the priority of relative/absolute path, executable on $PATH / user_local_path and so on.

Reply:
When this is delegated to the OS, then the OS needs to find it. And for a "normal" checker_bundler.exe, the Checker Library binary path needs to be in  $PATH (don't like).

So I propose something like
1. Try to find `program="..."` in the Checker Library directory
2. If not found, delegate to OS

Reply:
Hmm... I didn't mean it must be on $PATH. I agree that this shouldn't be mandatory.

However, the manifest will be defined by the checker library, right? So there is something like "configuration time". During this time the library knows verry well its location (via installation and automatic determination or via user's definition). Thus, it doesn't need to be on PATH, it is still defined (via absolute path), and we can delegate everything to OS. 

I'm thinking about three ways:

1. Developer:
E.g., I develop xosc_check.py. To test it I need to define the manifest. So I know verry well the configuration, e.g. `<Executable program="/home/user/.venv/dev/bin/python3"><Arg>/home/user/dev_project/xosc_check.py</Arg></Executable>`
This manifest will be somehow available to the framework (e.g. a search based process or registration).
2. Installation
The installation process will know where it placed exectuables, so the manifest is clear:
`<Executable program="python"><Arg>/usr/local/install_dir/xosc_check.py</Arg></Executable>`
3. The common "gity" way:
`git clone https://github.com/my-fancy-checker`
The checker should include a "deployment/installation" script or documentation:
`./install.sh` -> will place the executables and know the configuration.
or "Create a manifest in /x/y/z with the following content"

So at any time during this "configuration time" the execution method & call is clear and can be specified. Then the qc-framework has it easy at execution time since the configuration contains all this stuff.